### PR TITLE
Update proxy documentation

### PR DESCRIPTION
**Pull request by corbob**
## Description Of Changes

Update the proxy documentation to be clear about the order of proxy configurations applying. Add known issues with links.

## Motivation and Context

The proxy configuration has changed slightly from Chocolatey CLI 1.4 to 2.1.0

## Testing

* [x] I have previewed these changes using the [Docker Container](https://github.com/chocolatey/docs/tree/master/.devcontainer) or another method before submitting this pull request.

## Change Types Made

* [ ] Minor documentation fix (typos etc.).
* [x] Major documentation change (refactoring, reformatting or adding documentation to existing page).
* [ ] New documentation page added.
* [ ] The change I have made should have a video added, and I have raised an issue for this.
    * Issue #

## Change Checklist

* [ ] Requires a change to menu structure (top or left hand side)/
* [ ] Menu structure has been updated

## Related Issue

* ENGTASKS-3200

### CODE REVIEW

File: input/en-us/guides/usage/proxy-settings-for-chocolatey.md [hunk 1]
```diff
@@ -6,7 +6,26 @@ Description: Settings up Chocolatey to use locally configured proxy server
...
+1. [Chocolatey Configuration](#explicit-proxy-settings).
+1. [Environment Variables](#existing-proxy-environment-variables).
+1. [System configured proxy server](#system-proxy-settings).
+
+### Known Limitations
+
+Certain Chocolatey products are known to apply different proxy settings depending on circumstances.
+
+* Chocolatey GUI - [Downloading icons does not honor Chocolatey CLI proxy settings](https://github.com/chocolatey/ChocolateyGUI/issues/1013)
+* Chocolatey Licensed Extension - [VirusTotal checking at runtime does not use any proxy](https://gitlab.com/chocolatey/collaborators/choco-licensed/-/issues/681)
```
Comment (TheCakeIsNaOH) [inline input/en-us/guides/usage/proxy-settings-for-chocolatey.md:25]: This is not a publicly available issue, it probably should be linked/synced to the home repository.

Comment (corbob) [reply input/en-us/guides/usage/proxy-settings-for-chocolatey.md]: Thanks for noting that. These have been updated with public issues now.

File: input/en-us/guides/usage/proxy-settings-for-chocolatey.md [hunk 2]
```diff
@@ -6,7 +6,26 @@ Description: Settings up Chocolatey to use locally configured proxy server
 RedirectFrom: docs/proxy-settings-for-chocolatey
 ---
 
-## Installing Chocolatey from behind a proxy server
+## Proxy setting considerations
```
Comment (pauby) [inline input/en-us/guides/usage/proxy-settings-for-chocolatey.md:9]:
```suggestion
## Proxy Setting Considerations
```

Comment (pauby) [reply input/en-us/guides/usage/proxy-settings-for-chocolatey.md:9]: Please be consistent with headings.

File: input/en-us/guides/usage/proxy-settings-for-chocolatey.md [hunk 3]
```diff
@@ -6,7 +6,26 @@ Description: Settings up Chocolatey to use locally configured proxy server
...
+
+### Proxy Configuration Priority
+
+Chocolatey applies proxy configurations based on where they are supplied. The order of the configuration applying is:
```
Comment (pauby) [inline input/en-us/guides/usage/proxy-settings-for-chocolatey.md:13]:
```suggestion
Chocolatey applies proxy configuration based on the order below:
```

File: input/en-us/guides/usage/proxy-settings-for-chocolatey.md [hunk 4]
```diff
@@ -6,7 +6,26 @@ Description: Settings up Chocolatey to use locally configured proxy server
...
+
+1. [Command Line Arguments](#explicit-proxy-settings-at-runtime).
+1. [Chocolatey Configuration](#explicit-proxy-settings).
+1. [Environment Variables](#existing-proxy-environment-variables).
+1. [System configured proxy server](#system-proxy-settings).
```
Comment (pauby) [inline input/en-us/guides/usage/proxy-settings-for-chocolatey.md:18]:
```suggestion
1. [Command line arguments](#explicit-proxy-settings-at-runtime).
1. [Chocolatey config](#explicit-proxy-settings).
1. [Environment variables](#existing-proxy-environment-variables).
1. [System configured proxy server](#system-proxy-settings).
```

Comment (pauby) [reply input/en-us/guides/usage/proxy-settings-for-chocolatey.md:18]: Please be consistent.

File: input/en-us/guides/usage/proxy-settings-for-chocolatey.md [hunk 5]
```diff
@@ -6,7 +6,26 @@ Description: Settings up Chocolatey to use locally configured proxy server
...
+
+### Known Limitations
+
+Certain Chocolatey products are known to apply different proxy settings depending on circumstances.
+
+* Chocolatey GUI - [Downloading icons does not honor Chocolatey CLI proxy settings](https://github.com/chocolatey/ChocolateyGUI/issues/1013)
+* Chocolatey Licensed Extension - [VirusTotal checking at runtime does not use any proxy](https://github.com/chocolatey/chocolatey-licensed-issues/issues/351)
+* Chocolatey Agent - [Agent communications do not use any proxy configuration](https://github.com/chocolatey/chocolatey-licensed-issues/issues/350)
```
Comment (pauby) [inline input/en-us/guides/usage/proxy-settings-for-chocolatey.md:26]:
```suggestion
* Chocolatey GUI - [Downloading icons does not honor Chocolatey CLI proxy settings](https://github.com/chocolatey/ChocolateyGUI/issues/1013).
* Chocolatey Licensed Extension - [VirusTotal checking at runtime does not use any proxy](https://github.com/chocolatey/chocolatey-licensed-issues/issues/351).
* Chocolatey Agent - [Agent communications do not use any proxy configuration](https://github.com/chocolatey/chocolatey-licensed-issues/issues/350).
```

Comment (pauby) [reply input/en-us/guides/usage/proxy-settings-for-chocolatey.md:26]: Please be consistent.

File: input/en-us/guides/usage/proxy-settings-for-chocolatey.md [hunk 6]
```diff
@@ -6,7 +6,26 @@ Description: Settings up Chocolatey to use locally configured proxy server
...
+
+## Installing Chocolatey From Behind A Proxy Server
```
Comment (pauby) [inline input/en-us/guides/usage/proxy-settings-for-chocolatey.md:28]:
```suggestion
## Installing Chocolatey From Behind a Proxy Server
```

File: input/en-us/guides/usage/proxy-settings-for-chocolatey.md [hunk 7]
```diff
@@ -77,15 +96,15 @@ Chocolatey by default already supports system set proxy servers
...
-* `https_proxy`
-* `no_proxy`
+* `http_proxy`.
+* `https_proxy`.
+* `no_proxy`.
 
 ## Explicit Proxy Settings
 
-Chocolatey has explicit proxy support starting with 0.9.9.9.
+Chocolatey has explicit proxy
```
Comment (pauby) [inline input/en-us/guides/usage/proxy-settings-for-chocolatey.md:105]: This line looks incomplete?

File: input/en-us/guides/usage/proxy-settings-for-chocolatey.md [hunk 8]
```diff
@@ -77,15 +96,15 @@ Chocolatey by default already supports system set proxy servers
...
 
-You can simply configure 1 up to 5 settings and Chocolatey will use a proxy server. `proxy` is required and is the location and port of the proxy server. The values for user/password are only used for credentials when both are present.
+You can simply configure up to 5 settings and Chocolatey will use a proxy server. `proxy` is required and is the location and port of the proxy server. The values for user/password are only used for credentials when both are present.
 
 ~~~sh
 choco config set --name proxy --value <locationandport>
```
Comment (pauby) [inline input/en-us/guides/usage/proxy-settings-for-chocolatey.md:110]: Can you fix up all of this code based on best practices? Can you also capitalize the parts in the `<` brackets, for example `<LOCATION-AND-PORT>`?

File: input/en-us/guides/usage/proxy-settings-for-chocolatey.md [hunk 9]
```diff
@@ -97,7 +116,7 @@ choco config set --name proxyBypassOnLocal --value true # optional
 
 ### Example
 
-Running the following commands in 0.9.9.9:
+Running the following commands:
 
 ~~~sh
```
Comment (pauby) [inline input/en-us/guides/usage/proxy-settings-for-chocolatey.md:121]: Can you fixup this code too?

**Review by pauby — CHANGES_REQUESTED** [review]

File: input/en-us/guides/usage/proxy-settings-for-chocolatey.md (hunk 2 above)
Comment (corbob) [reply input/en-us/guides/usage/proxy-settings-for-chocolatey.md:9]: 😂 I explicitly added a maint commit to bring the headings into consistency, and then committed this one without it 😂. I've gotten that fixed up.

**Comment by corbob** [comment]
I've updated the docs with the requested changes from pauby. I have included them as a `fixup!` commit with the intention that we should `git rebase upstream/master -i --autosquash` before the merge which will bring the branch in line with `master`, as well as fixup the mentioned commit into the original commit where the changes were made.

**Review by vexx32 — APPROVED** [review]
This looks good to me, thanks for fixing this up! 💖

**Pull request merged**